http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/LeoPublisher.scala

```diff
@@ -0,0 +1,56 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import cats.effect.{Sync, Timer}
+import cats.implicits._
+import com.google.protobuf.ByteString
+import com.google.pubsub.v1.PubsubMessage
+import fs2.concurrent.InspectableQueue
+import fs2.{Pipe, Stream}
+import io.chrisdavenport.log4cats.Logger
+import io.circe.syntax._
+import org.broadinstitute.dsde.workbench.google2.GooglePublisher
+import org.broadinstitute.dsde.workbench.leonardo.db.{clusterQuery, DbReference}
+import org.broadinstitute.dsde.workbench.leonardo.http.{dbioToIO, nowInstant}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubCodec._
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage
+
+final class LeoPublisher[F[_]: Logger: Timer](
```

Comment:
Can you add a comment about what this class is for? I think it's used in the front-leo `service` layer to publish messages to PubSub, right?

Can we put it in `service` package instead of top-level `leonardo` package?

Reply:
it basically provides a publishing stream (`process`)...It's not being used in anywhere else other than `boot`...

Reply:
I see.. it's not called directly by the `service` classes. The current location next to `Boot` is fine I guess.

Reply:
yea, so this is just for dequeueing from publisherQueue and transform it into a native google pubsub message; After pubsub message is published, we update database when necessary..I added this to comment...

And services that need to publish event, they do so by enqueue a message into the internal publisherQueue (a fs2.Queue)